public/static/js/spawned_hosts.js

```diff
@@ -252,99 +266,100 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope','$window', '$timeout', 'mciSp
-                },
-                error: function (resp) {
-                    notificationService.pushNotification('Error updating host tags: ' + resp.data.error,'errorHeader');
-                }
+          'updateHostTags',
+          $scope.curHostData.id, tags_to_add, $scope.curHostData.tags_to_delete, {}, {
+            success: function (resp) {
+              window.location.href = "/spawn";
+            },
+            error: function (resp) {
+              notificationService.pushNotification('Error updating host tags: ' + resp.data.error, 'errorHeader');
             }
+          }
         );
-    };
-
-    $scope.updateInstanceType = function() {
-      // Do nothing if no instance type selected
-      if (!$scope.curHostData.selectedInstanceType) {
-        return
       }
-      // Do nothing if host is not stopped
-      if ($scope.curHostData.status != "stopped") {
-        notificationService.pushNotification('Host must be stopped before modifying instance type', 'errorHeader');
-        return
-      }
-      mciSpawnRestService.updateInstanceType(
-        'updateInstanceType',
-        $scope.curHostData.id, $scope.curHostData.selectedInstanceType, {}, {
-          success: function(resp) {
-            window.location.href = "/spawn";
-          },
-          error: function(resp) {
-            notificationService.pushNotification('Error setting new instance type: ' + resp.data.error, 'errorHeader')
-          }
+
+      // update instance type if it changed
+      if ($scope.curHostData.selectedInstanceType && $scope.curHostData.instance_type !== $scope.curHostData.selectedInstanceType) {
+        // Do nothing if host is not stopped
+        if ($scope.curHostData.status != "stopped") {
+          notificationService.pushNotification('Host must be stopped before modifying instance type', 'errorHeader');
+          return
         }
-      )
+        mciSpawnRestService.updateInstanceType(
+          'updateInstanceType',
+          $scope.curHostData.id, $scope.curHostData.selectedInstanceType, {}, {
+            success: function (resp) {
+              window.location.href = "/spawn";
+            },
+            error: function (resp) {
+              notificationService.pushNotification('Error setting new instance type: ' + resp.data.error, 'errorHeader')
+            }
+          }
+        );
+      }
     }
 
-    $scope.updateHostStatus = function(action) {
+    $scope.updateHostStatus = function (action) {
       mciSpawnRestService.updateHostStatus(
         action,
         $scope.curHostData.id, {}, {
-          success: function(resp) {
+          success: function (resp) {
             window.location.href = "/spawn";
           },
-          error: function(jqXHR, status, errorThrown) {
-            notificationService.pushNotification('Error changing host status: ' + resp.data.error,'errorHeader');
+          error: function (jqXHR, status, errorThrown) {
```

Comment:
I know you didn't write this but what's up with this function signature

Reply:
Why is it `(jqXHR, status, errorThrown)` when the pushNotification uses resp.data? Should this not still be `function(resp)` ?

Reply:
Oh whoa that's just a bug. I fixed it